webcompat/static/js/lib/bugform.js

```diff
@@ -77,23 +76,34 @@ function BugForm() {
     // Set up listener for message events from screenshot-enabled add-ons
     window.addEventListener('message', _.bind(function(event) {
       // Make sure the data is coming from ~*inside the house*~!
-      // (i.e., our add-on sent it)
+      // (i.e., our add-on or some other priviledged code sent it)
       if (location.origin === event.origin) {
-        this.screenshotData = event.data;
-
-        // The final size of Base64-encoded binary data is ~equal to
-        // 1.37 times the original data size + 814 bytes (for headers).
-        // so, bytes = (encoded_string.length - 814) / 1.37)
-        // see https://en.wikipedia.org/wiki/Base64#MIME
-        if ((String(this.screenshotData).length - 814 / 1.37) > this.UPLOAD_LIMIT) {
-          this.downsampleImageAndUpload(this.screenshotData);
+        // See https://github.com/webcompat/webcompat.com/issues/1252 to track
+        // the work of only accepting blobs, which should simplify things.
+        if (event.data instanceof Blob) {
+          // showUploadPreview will take care of converting from blob to
+          // dataURI, and will send the result to resampleIfNecessaryAndUpload.
+          this.showUploadPreview(event.data);
```

Comment:
I am a bit confused that a function called `showUploadPreview` ultimately triggers the upload itself, since I would normally not expect that. But changing that in this pull request is probably not in scope, and probably not even needed.